include/quicly/cc.h

```diff
@@ -85,23 +148,63 @@ typedef struct st_quicly_cc_t {
     uint32_t num_loss_episodes;
 } quicly_cc_t;
 
+struct st_quicly_cc_impl_t {
+    void (*quicly_cc_init)(quicly_cc_t *, const quicly_cc_conf_t *, uint32_t);
+    void (*quicly_cc_on_acked)(quicly_cc_t *, const quicly_loss_t *, uint32_t, uint64_t, uint32_t, uint32_t);
```

Comment:
I would have preferred to use `const quicly_conn_t *` directly to make the interface more generic, but the struct is only defined in quicly.c. It would also allow the removal of the `max_udp_payload_size` (last uint32) parameter.

I understand that it shouldn't be exposed in a public header, but maybe defining it in a header next to the source files (for inclusion via `#include "..."`) would be a solution to this?

Reply:
I think the other question here is if we want to have unit tests for CC. If the answer is yes, then not having dependency from CC to `quicly_conn_t` is highly beneficial.

I think that having no unit test for Reno / Cubic is fine, but the importance of unit tests rise as the we add complex CC. For that, I might prefer retaining isolation.

Reply:
Good point, I hadn't actually considered unit tests. 

Adding more parameters to the functions in the future is pretty simple, and certainly not everything in `quicly_conn_t` is relevant for CC. That raises the question of what level of abstraction to choose though, e.g., CUBIC only uses the RTT estimate from `quicly_loss_t`. I chose to include the entire struct since other CCs might use different parts.

Reply:
I agree that the level of abstraction is the concern. That would continue to be, as we change things. Eventually, we might end up in having an set of callbacks that provide the necessary information.

But for the time being I think what we have in the PR is good.